.github/workflows/main.yml

```diff
@@ -147,3 +148,26 @@ jobs:
     - run: ci/fuzzit.sh fuzzing
       env:
         FUZZIT_API_KEY: ${{ secrets.FUZZIT_API_KEY }}
+
+  crates_io_publish:
+    needs:
+      - rustfmt
+      - docs
+      - test
+    name: Publish to crates.io
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Install Rust
+      run: rustup update stable && rustup default stable
+
+    - name: Configure env vars
+      run: |
+        echo ::set-env name=NO_DRY_RUN::1
+        echo ::set-env name=CARGO_REGISTRY_TOKEN::${{ secrets.CARGO_REGISTRY_TOKEN }}
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+
+    - name: Publish
+      run: rustc ci/publish.rs && ./publish publish
```

Comment:
Do we need to do anything special to tell it to only publish if this is the master branch, and not some random feature branch?

Reply:
This'll only actually do things for tags (previous step enables), and only for tags that start with a `v` (I'm not great on regexes). Do you think it should be limited further though?